### PR TITLE
DDF-2264 Upgrade Camel to 2.16.3 and add support for camel-amqp

### PR DESCRIPTION
**Pull request by roelens8**
#### What does this PR do?
Upgrade camel to 2.16.3 and adds the ability to install the camel-amqp component.
#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged; if a component team is listed, at least one of its members needs to approve)?
@ryeats @coyotesqrl
#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@kcwire 
@jaymcnallie
#### How should this be tested?
Make sure integration tests pass and that the camel-amqp feature can install correctly.
#### What are the relevant tickets?
DDF-2264
#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

**Comment by ryeats** [comment]
👍  LGTM

File: platform/platform-app/src/main/resources/features.xml [hunk 1]
```diff
@@ -53,128 +53,135 @@
...
+    </feature>
+    <feature name='camel-amqp' version='${project.version}' start-level='50'>
+    <bundle dependency='true'>mvn:commons-lang/commons-lang/${commons-lang.version}</bundle>
+    <bundle dependency='true'>mvn:commons-collections/commons-collections/${commons-collections.version}</bundle>
+    <bundle dependency='true'>mvn:org.apache.servicemix.bundles/org.apache.servicemix.bundles.mina/${servicemix.bundles.mina.version}</bundle>
+    <!-- TODO: Specify version range for qpid-jms-client (0.6.0, 0.10.0] once Karaf 4.0.6 is available -->
+    <bundle dependency='true'>wrap:mvn:org.apache.qpid/qpid-jms-client/${qpid-jms-client.version}</bundle>
+    <bundle dependency='true'>wrap:mvn:org.apache.qpid/proton-j/${proton-j.version}</bundle>
+    <bundle dependency='true'>wrap:mvn:io.netty/netty-all/${netty-all.version}</bundle>
+    <feature prerequisite="true" version='${camel.version}'>camel-jms</feature>
```
Comment (coyotesqrl) [inline platform/platform-app/src/main/resources/features.xml]: Nit, but I prefer to see the dependent bundles and prereq features at the top of the feature definition with those that it installs beneath. I believe it reads more clearly.

**Comment by coyotesqrl** [comment]
:+1:

File: platform/platform-app/src/main/resources/features.xml (hunk 1 above)
Comment (stustison) [inline platform/platform-app/src/main/resources/features.xml]: Assuming he did it exactly how I did it previously, these should all be copy and pasted straight out of the Camel features file.

File: pom.xml [hunk 2]
```diff
@@ -97,6 +97,16 @@
         <pax.cdi.version>0.12.0</pax.cdi.version>
         <validation.version>1.1.0.Final</validation.version>
         <commons-configuration.version>1.10</commons-configuration.version>
+        <servicemix.bundles.mina.version>1.1.7_6</servicemix.bundles.mina.version>
```
Comment (stustison) [inline pom.xml]: Are all of these new version variables in the copy/pasted features file? If not, there is a good chance that they'll just become stale when we update camel again. Your best bet is to just copy and paste the features straight from the file with no modification. We should have comments in the features file stating as such.

File: platform/platform-app/src/main/resources/features.xml (hunk 1 above)
Comment (roelens8) [inline platform/platform-app/src/main/resources/features.xml]: That's correct, these features are copied straight from the camel features file.

File: pom.xml (hunk 2 above)
Comment (roelens8) [inline pom.xml]: All of these version variables are in the copy/pasted features file and are being used. I don't see these version variables becoming stale unless we upgrade to newer a version of camel that doesn't use a particular dependency that we have defined a version variable for. In that case, we just have to remember to remove it.

**Comment by kcwire** [comment]
👍

**Comment by jaymcnallie** [comment]
Bamboo build keeps throwing `OutOfMemoryError` after the build completes successfully.  Anybody know if this is a byproduct of the change or bamboo crumbling under load?  I've run the full build successfully locally several times with no issue.

**Comment by coyotesqrl** [comment]
It's just Bamboo. Those server instances just aren't big enough for our builds to spit out logs much of the time. One of the many reasons Atlassian is giving up Bamboo in the cloud.

**Pull request merged**